src/CourseAuthoringPage.jsx

```diff
@@ -45,10 +48,10 @@ export default function CourseAuthoringPage({ courseId, children }) {
       <Footer />
     </div>
   );
-
+  const { pathname } = useLocation();
   return (
     <div className="bg-light-200">
-      {inProgress ? <Loading /> : <AppHeader />}
+      {inProgress ? !pathname.includes('/editor/') && <Loading /> : <AppHeader />}
```

Comment:
What are the implications of this hard-coding of `/editor/` in the path? Should this path be split out into a separate constant? Does it need to be kept in-sync with a path elsewhere in the code? If so, shouldn't a comment be added to that effect?

Reply:
Great question, I see what you're getting at here, and I'm not sure of an elegant solution, so I'm just going to offer up what I know and dream up one. The pathname comes from the hook to the React router `history` object, so in theory, any change in the pathname will cause a re-render, and that conditional will be evaluated. In terms of the problem of making sure any change to the url is reflected in here, perhaps we could have both the CourseAuthoringRoutes component and this check here reference the same value exported from the CourseAuthoringRoutes.jsx file, so that if that ever were to change in that location, it would be reflected here? I don't know what would be best, because that adds another layer of complexity. The tradeoff is also informed by the fact that this is a temporary thing, as it won't always be served as a "flattened" modal in future, so maybe just a comment to make sure to remove this check is in order.

Reply:
I've added a task [task](https://openedx.atlassian.net/browse/TNL-9591) and a comment to reflect the current state. Tests will fail if this is ever broken, which is nice.

Reply:
A created ticket ✅  and a comment ✅ were exactly what I was looking for here.